Mario tests

marioney opened the pull request:
Update files and instructions to launch the Pilot on a simulated TB3

jginesclavero commented on README.md:
````diff
@@ -37,17 +37,57 @@ The modes change next parameters in the components (shown in the figure), with s
+
+  ```console
+    source ~/navigation2_ws/install/setup.bash
+    mkdir -p ~/turtlebot3_ws/src
````
IMHO, we must have one WS with all necessary packages, instead of three WS.

marioney replied:
We need two separate WS to compile the BT and the Navigation, if we use the binaries, that will not be the case.  The third WS was for TB3, but it can of course be on the same WS as the Nav2.

marioney commented on README.md:
```diff
@@ -37,17 +37,57 @@ The modes change next parameters in the components (shown in the figure), with s
+    wget https://raw.githubusercontent.com/MROS-RobMoSys-ITP/Pilot-URJC/mario-tests/nav2_dependencies.repos
+    vcs import src < nav2_dependencies.repos
+    rosdep install -y -r -q --from-paths src --ignore-src --rosdistro eloquent
+    colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
```
> Why do you import a specific commit of Behaviortrees?

It didn't compile with other commits of the BT, see https://github.com/MROS-RobMoSys-ITP/Pilot-URJC/issues/20#issuecomment-645951383

> we must use the BT from debian packages and it is installed with the next rosdep install.

I agree, that will be the best way, but there were some errors while building the BICA package when using that. 

Note:  I just saw that you have updated the BICA package https://github.com/IntelligentRoboticsLabs/BICA/commit/2fc1a497a353b10a4f0f5a7f489f13c1a16f8c95  So I'll test the new version

chcorbato replied:
> I agree, that will be the best way, but there were some errors while building the BICA package when using that.
> 
> Note: I just saw that you have updated the BICA package [IntelligentRoboticsLabs/BICA@2fc1a49](https://github.com/IntelligentRoboticsLabs/BICA/commit/2fc1a497a353b10a4f0f5a7f489f13c1a16f8c95) So I'll test the new version

@fmrico fixed those in that update, at least it worked for me.

fmrico commented:
Hi @jginesclavero and @marioney 

We have to avoid using a specific version of packages. We have to use binary packages or master. The (only) exception is Navigation2, because we are contributing to the source code of this package, and master is for Foxy. That's why we should use the last release (0.3.4, I think) for eloquent.

If we do many hacks for running the software, it will be very hard for anyone else to reproduce our results. We also should use only one workspace.

The problem with BICA is solved. We should use `master` for BICA and PlanSys2 (we could even use binary package for this).

Best

marioney commented:
> The problem with BICA is solved. We should use `master` for BICA

Maybe I'm wrong, but doesn't we need the `behavior_trees` branch for BICA

fmrico commented:
@marioney You are (almost) right. `master` is for ROS1. We should use `ros2` branch, that is synchronized with `behavior trees`

marioney commented:
I've updated the instructions and the `.repos` files to work with installed versions of BT and PlanSys.

I've also re-based this branch to take in the latest changes to `master` 

There's still one compilation error on BICA/bica_examples (see https://github.com/IntelligentRoboticsLabs/BICA/issues/14) but I think it should not affect this. (I just added a CATKIN_IGNORE on my local copy)

chcorbato commented:
> I've updated the instructions and the `.repos` files to work with installed versions of BT and PlanSys.
> 
> I've also re-based this branch to take in the latest changes to `master`

Great! I'll check asap.

> There's still one compilation error on BICA/bica_examples (see [IntelligentRoboticsLabs/BICA#14](https://github.com/IntelligentRoboticsLabs/BICA/issues/14)) but I think it should not affect this. (I just added a CATKIN_IGNORE on my local copy)

We should get rid of all errors. I hope someone will be able to figure this one out too.

fmrico commented on dependencies.repos:
```diff
@@ -1,14 +1,17 @@
-    url: https://github.com/IntelligentRoboticsLabs/ros2_planning_system.git
-    version: master
   
+  Pilot-URJC:
```
@marioney I think that we shouldn't include as dependence the repo itself

marioney replied:
OK, I can change that. I just put it that way to have one less instruction.

marioney replied:
Solved in 77c43904922b516800c00c4ca219c4eec087f001

fmrico commented on pilot_urjc_bringup/maps/sim/tb3_world/map.yaml:
```diff
@@ -0,0 +1,7 @@
+image: /home/mario/map.pgm
```
This image path probably is not valid

marioney replied:
I'll change that - copy/paste error I guess

marioney replied:
Solved in 062f8f9ac25701a33de5b9836094f1808d559c49

fmrico commented on pilot_urjc_bringup/launch/turtlebot3_sim/nav2_turtlebot3_launch.py:
```diff
@@ -0,0 +1,196 @@
+# Copyright (c) 2018 Intel Corporation
```
Why copy/paste the original launcher in nav2 instead of including it?

marioney replied:
I just did the same as that was there for the other two launchers [nav2_tiago_launch.py}(https://github.com/MROS-RobMoSys-ITP/Pilot-URJC/blob/master/pilot_urjc_bringup/launch/tiago/nav2_tiago_launch.py) and [nav2_turtlebot2_launch.py](https://github.com/MROS-RobMoSys-ITP/Pilot-URJC/blob/master/pilot_urjc_bringup/launch/turtlebot2_sim/nav2_turtlebot2_launch.py)

All of them have some differences from the original launcher in nav2, they launch the system modes and actually call a `nav2_bringup_launch` that's inside the `pilot_urjc_bringup`

fmrico commented:
Hi @marioney 

I have just pushed a commit that solves the compilation problem with BTs, as far as I could test it.

I used:
* behaviortree-cpp-v3 installed by packages with an updated eloquent distribution.
* BICA (branch: `ros2`)
* Pilot-URJC  (branch: `master`)
* mros_navigation2 (beach: `master`)
* ros2_planning_system (beach: `master`)

Please, confirm also that this is working for you.

Best

marioney commented:
> Please, confirm also that this is working for you.

I still get the error: `1 package failed: bica_examples` it's the same error I reported before (https://github.com/IntelligentRoboticsLabs/BICA/issues/14)

fmrico commented:
Yes, you will have to use plansys2 from sources until next version will be delivered in next eloquent sync. Remove the plansys2 installation from binary packages.

marioney commented:
>  Remove the plansys2 installation from binary packages.

That solves it.  It's compiling now without any issues :+1:

chcorbato commented:
> Please, confirm also that this is working for you.

Nothing better than building your ws without errors and running a nice robot simulation first thing in the morning, thanks @fmrico @marioney @jginesclavero 👏 👏

fmrico commented:
Ok, LGTM

I am merging, as @jginesclavero is out these days and I don't want to delay this PR.

fmrico commented:
Thanks, Mario!!!
